### PR TITLE
fix: Sinkhorn OT 4-arg API, trajectory B-factors, wandb global_step, …

### PR DESCRIPTION
**Pull request by alisiafadini**
…a couple random bits unfortunately

- realspace.py: Fix Sinkhorn loss to use 4-argument SamplesLoss API (alpha, x, beta, y) with proper density weighting; add configurable sinkhorn_density_threshold and sinkhorn_blurs parameters (issue #7 to keep in mind)

- trajectory.py: Pass B-factors to PDB writer (clamped 0-99.99), remove conflicting per-frame wandb.log that fought with engine step counter, add canonical best.pdb copy

- engine.py: Add monotonic _global_step for wandb across multi-run refinements, plumb lr_scheduler through run/optimize, log LR and run_id to wandb

- map_utils.py: Fix tv_denoise_difference_map unpacking (meteor API)

### CODE REVIEW

File: LossLab/losses/realspace.py [hunk 1]
```diff
@@ -49,6 +49,8 @@ def __init__(
         loss_type: Literal["cc", "l2", "sinkhorn", "density_explained"] = "l2",
         mask_center: np.ndarray | None = None,
         mask_radius: float | None = None,
+        sinkhorn_density_threshold: float = 0.0,
+        sinkhorn_blurs: list[float] | None = None,
```
Comment (tjlane) [inline LossLab/losses/realspace.py]: document these mysterious args in the docstring
